File: javascript/lasagna/lasagna.js
// @ts-check

/**
 * The number of minutes it takes to prepare a single layer.
 */
const MINUTES_PER_LAYER = 2;
const EXPECTED_MINUTES_IN_OVEN = 40;

/**
 * Determines the number of minutes the lasagna still needs to remain in the
 * oven to be properly prepared.
 *
 * @param {number} minutes
 * @returns {number} the number of minutes remaining
 */
export function remainingMinutesInOven(minutes) {
  return EXPECTED_MINUTES_IN_OVEN - minutes;
}

/**
 * Given a number of layers, determines the total preparation time.
 *
 * @param {number} layers
 * @returns {number} the total preparation time
 */
export function preparationTimeInMinutes(layers) {
  return MINUTES_PER_LAYER * layers;
}

/**
 * Calculates the total working time. That is, the time to prepare all the layers
 * of lasagna, and the time already spent in the oven.
 *
 * @param {number} layers
 * @param {number} minutes
 * @returns {number} the total working time
 */
export function totalTimeInMinutes(layers, minutes) {
  return minutes + preparationTimeInMinutes(layers);
}

export { EXPECTED_MINUTES_IN_OVEN };

File: python/guidos-gorgeous-lasagna/lasagna.py
"""
Solution to lasagna exercise.
"""

EXPECTED_BAKE_TIME = 40
PREPARATION_TIME = 2


def bake_time_remaining(elapsed_bake_time: int) -> int:
    """
    Return the bake time remaining.

    Return the amount of time remaining to bake the lasagna given how long it's been baking.
    """
    return EXPECTED_BAKE_TIME - elapsed_bake_time


def preparation_time_in_minutes(layers: int) -> int:
    """
    Return the prep time in minutes.

    Return how much time it will take to prepare the lasagna for baking given how many layers.
    are required.
    """
    return PREPARATION_TIME * layers


def elapsed_time_in_minutes(number_of_layers: int, elapsed_bake_time: int) -> int:
    """
    Return elapsed cooking time.

    This function takes two numbers representing the number of layers & the time already spent
    baking and calculates the total elapsed minutes spent cooking the lasagna.
    """
    return elapsed_bake_time + preparation_time_in_minutes(number_of_layers)
